DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Centre/Administrator/AdministratorControllerTests.cs

```diff
@@ -188,5 +202,163 @@ public void UnlockAccount_unlocks_account_and_returns_to_page()
             A.CallTo(() => userDataService.UpdateAdminUserFailedLoginCount(1, 0)).MustHaveHappened();
             result.Should().BeRedirectToActionResult().WithActionName("Index");
         }
+
+        [Test]
+        public void EditAdminRoles_edits_roles_when_spaces_available()
+        {
+            // Given
+            var currentAdminUser = UserTestHelper.GetDefaultAdminUser(
+                isContentCreator: false,
+                isTrainer: false,
+                importOnly: false,
+                isContentManager: false
+            );
+            var newRoles = EditAdminRolesTestHelper.GetDefaultEditRolesViewModel(
+                1,
+                true,
+                true,
+                true,
+                true,
+                ContentManagementRole.CmsAdministrator
+            );
+            var numberOfAdmins = NumberOfAdministratorsTestHelper.GetDefaultNumberOfAdministrators();
+            A.CallTo(() => userDataService.GetAdminUserById(A<int>._)).Returns(currentAdminUser);
+            A.CallTo(() => numberOfAdministratorsService.GetCentreAdministratorNumbers(A<int>._))
+                .Returns(numberOfAdmins);
+
+            // When
+            var result = administratorController.EditAdminRoles(newRoles, currentAdminUser.Id);
+
+            // Then
+            A.CallTo(

```

Comment:
These tests are quite long and hence more difficult to glance over. One way to make them a bit shorter and better readable would be to pull some stuff into private methods. What I can think of atm is that we could pull this method into AssertAdminPermissionsUpdatedCorrectly(id, newRoles). As a bonus, this can then be reused for the test below.